### ISSUE TITLE
function isExcluded in noUnusedRule.ts: invalid condition

### ISSUE THREAD

**Issue by m-andresen** [ajafff/tslint-consistent-codestyle#31]
File: [../blob/master/rules/noUnusedRule.ts](../blob/master/rules/noUnusedRule.ts)
```javascript
172    switch (parent.kind) {
180    case ts.SyntaxKind.BindingElement:
184        if (pattern.kind === ts.SyntaxKind.ObjectBindingPattern &&
```
The condition in line 184 never true.

Example:
```javascript
    @Effect()effect  onSetStep = this.action$
        .ofType<SwitchAction>(SwitchAction.TYPE)
        .withLatestFrom(this.stepbarService.stepbar$)
        .map(([_action, stepbar]) => { // the _action-Variable is correctly prefixed but marked by the lint
            return new Action(stepbar.item);
        });
```

**Comment by ajafff** [comment]
That's actually expected. It's also documented: https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-unused.md#ignoring-uninteresting-parameters-and-variables (note that array destructuring is not mentioned in the list)

The rationale behind this: In array destructuring you can omit values at certain positions.
In your case it looks like this:
```ts
    @Effect()effect  onSetStep = this.action$
        .ofType<SwitchAction>(SwitchAction.TYPE)
        .withLatestFrom(this.stepbarService.stepbar$)
        .map(([, stepbar]) => { // note the removed variable _action
            return new Action(stepbar.item);
        });
```

**Comment by m-andresen** [comment]
Ok, it's true, you can omit the _action-Value. I didn't know that before.

Nevertheless the condition in line 184 is never fulfilled and therefore line 186 `return true` is never executed:
```javascript
console.log('ts.SyntaxKind.BindingElement=', ts.SyntaxKind.BindingElement);
// output: ts.SyntaxKind.BindingElement=176
console.log('ts.SyntaxKind.ObjectBindingPattern=', ts.SyntaxKind.ObjectBindingPattern);
// output: ts.SyntaxKind.ObjectBindingPattern=174

172    switch (parent.kind) {
180    case ts.SyntaxKind.BindingElement:
184        if (pattern.kind === ts.SyntaxKind.ObjectBindingPattern && // <- false
185            pattern.elements[pattern.elements.length - 1].dotDotDotToken !== undefined)
186            return true
```

**Comment by ajafff** [comment]
It's hard to see, but these are two different variables: `parent !== pattern`

**Comment by ajafff** [comment]
You can also see that the code path is hit by the tests: https://coveralls.io/builds/12951073/source?filename=rules%2FnoUnusedRule.ts#L180

**Comment by m-andresen** [comment]
thank you, you are right
